### DISCUSSION TITLE
Are there plans to introduce error codes to the parse and evaluation exceptions?

### DISCUSSION THREAD

**Discussion by omniavincitlabor** [ezylang/EvalEx#404] [Q&A]
Hello,

i really like the library and especially the new features.

So I am currently trying to get rid of my old and  "highly customized" version (meaning tons of very ugly code) and update to the new EvalEx-version. My last open issue is translating the exception messages into german for display in the UI. For this purpose I had error codes in my "customization".

So my question is:
Are there any plans to have error codes in ParseException and EvaluationException?

**Comment by uklimaschewski** [comment]
No plans there, yet. Though now with the more cleaned up exception handling, it would  easier to add error codes.
Maybe something for 3.2 or later.

**Comment by RobertZenz** [comment]
Given that the possible exception messages are finite, you could always use the English messages as keys. Some of them might contain some additional information, but nothing a fuzzy matcher can't solve, would also make your translation file a little bit better readable.

```
"Missing second operand for operator": "Zweiter Operand fehlt fuer diese Operation"
"Missing operator": "Fehlender Operator"
```
